### DISCUSSION TITLE
Configuration directory ignored?

### DISCUSSION THREAD

**Discussion by roukmoute** [vimeo/psalm#7823] [Q&A]
The first example [here](https://psalm.dev/docs/running_psalm/configuration/) consider `directory name="src"`.

But when I run it, I see `vendor/`, `public` in errors, so other folders than the one defined in the configuration file.
So what is the point of defining `<directory name="src" />`?

Thanks in advance

**Comment by orklah** [comment]
src is the directory you want to analyse code on. However, Psalm will need to scan function definitions on your vendors (so it know that $vendorObject->AFunction() returns a string and take no argument)

So, on a normal day, Psalm *scans* /vendor and *analyze* /src. However, there are cases where Psalm encounters some alien declaration on your vendors and Psalm can't scan properly. Depending on the issue, Psalm can crash or emit an issue.

However, Psalm won't go inside the functions in /vendor to check for errors there, it's merely retrieving declaration to be able to do its job properly

**Reply by roukmoute** [reply]
Ok thanks!

I think I understand better.

But with your last sentence, shouldn't he ignore what he is currently returning to me?

Here is an example (Symfony skeleton 4.4) :  

```sh
ERROR: UnresolvableInclude - src/Kernel.php:20:21 - Cannot resolve the given expression to a file path (see https://psalm.dev/106)
        $contents = require $this->getProjectDir().'/config/bundles.php';

ERROR: MissingReturnType - vendor/symfony/framework-bundle/Kernel/MicroKernelTrait.php:57:21 - Method Symfony\Bundle\FrameworkBundle\Kernel\MicroKernelTrait::registerContainerConfiguration does not have a return type, expecting void (see https://psalm.dev/050)
    public function registerContainerConfiguration(LoaderInterface $loader)

ERROR: MissingReturnType - vendor/symfony/framework-bundle/Kernel/MicroKernelTrait.php:90:21 - Method Symfony\Bundle\FrameworkBundle\Kernel\MicroKernelTrait::loadRoutes does not have a return type, expecting Symfony\Component\Routing\RouteCollection (see https://psalm.dev/050)
    public function loadRoutes(LoaderInterface $loader)

ERROR: DeprecatedProperty - vendor/symfony/http-kernel/Kernel.php:93:9 - Symfony\Component\HttpKernel\Kernel::$rootDir is marked deprecated (see https://psalm.dev/099)
        $this->rootDir = $this->getRootDir(false);

ERROR: DeprecatedMethod - vendor/symfony/http-kernel/Kernel.php:93:33 - The method Symfony\Component\HttpKernel\Kernel::getRootDir has been marked as deprecated (see https://psalm.dev/001)
        $this->rootDir = $this->getRootDir(false);

...
```

**Reply by orklah** [reply]
There may indeed be an exception when a class in your code extends another class in vendors. Psalm will sometimes need to analyse the parent class to some degree for specific checks.

I don't have everything in memory, but for example, if you call a parent method in your constructor, Psalm will need to check that function in order to see if the parent has initialized properties in your child.

In this case, I would expect that excluding Kernel.php should solve your issues

**Reply by roukmoute** [reply]
Yes, that's what I had done, but I wanted to ask to better understand what's going on.
I thank you for your help.
